staging/kos/pkg/controllers/ipam/ipam.go

```diff
@@ -780,18 +780,26 @@ func (ctlr *IPAMController) pickAndLockAddress(ns, name string, att *netv1a1.Net
 }
 
 func (ctlr *IPAMController) updateNAStatus(ns, name string, att *netv1a1.NetworkAttachment, nadat *NetworkAttachmentData, statusErrs []string, subnetUID k8stypes.UID, lockForStatus ParsedLock, ipForStatus gonet.IP) error {
-	att2 := att.DeepCopy()
-	att2.Status.Errors.IPAM = statusErrs
-	att2.Status.LockUID = string(lockForStatus.UID)
-	att2.Status.AddressVNI = lockForStatus.VNI
+	var pb strings.Builder
+	pb.WriteString(`{"status": {"errors": {"ipam": [`)
```

Comment:
Why not doing what is done by the CA, that is, initializing a `NetworkAttachmentStatus` struct and serializing it to JSON? It's more elegant than dealing directly with strings.

Reply:
Yes, and no (due to `omitempty`).